tortoise/backends/base/schema_generator.py

```diff
@@ -449,16 +364,104 @@ def _get_table_sql(self, model: "type[Model]", safe: bool = True) -> dict:
+        for field_name, column_name in model._meta.fields_db_projection.items():
+            field_object = model._meta.fields_map[field_name]
+            comment = self._generate_field_comment(field_object, table_name, column_name)
+            default = self._generate_field_default(field_object, table_name, column_name, model)
+
+            # TODO: PK generation needs to move out of schema generator.
+            if create_pk_field := self._generate_pk_create_sql(field_object, column_name, comment):
+                fields_to_create.append(create_pk_field)
+                continue
+
+            field_creation_string, related_table_name = self._generate_field_creation(
+                field_object, table_name, column_name, default, comment
+            )
+            if related_table_name:
+                references.add(related_table_name)
+            fields_to_create.append(field_creation_string)
+
+            if field_object.index and not field_object.pk:
+                fields_with_index.append(column_name)
+
+        if model._meta.unique_together:
+            for unique_together_list in model._meta.unique_together:
+                unique_together_to_create = [
+                    model._meta.fields_map[field].source_field or field
+                    for field in unique_together_list
+                ]
+                fields_to_create.append(
+                    self._get_unique_constraint_sql(model, unique_together_to_create)
+                )
+
+        field_indexes_sqls = self._generate_indexes(model, fields_with_index, safe)
+
+        fields_to_create.extend(self._get_inner_statements())
+
+        table_fields_string = "\n    {}\n".format(",\n    ".join(fields_to_create))
+        table_comment = (
+            self._table_comment_generator(table=table_name, comment=model._meta.table_description)
+            if model._meta.table_description
+            else ""
+        )
+
+        table_create_string = self.TABLE_CREATE_TEMPLATE.format(
+            exists="IF NOT EXISTS " if safe else "",
+            table_name=table_name,
+            fields=table_fields_string,
+            comment=table_comment,
+            extra=self._table_generate_extra(table=table_name),
+        )
+
+        table_create_string = "\n".join([table_create_string, *field_indexes_sqls])
+
+        table_create_string += self._post_table_hook()
+
+        m2m_tables_for_create = self._generate_m2m_tables(model, table_name, safe, models_tables)
 
         return {
-            "table": model._meta.db_table,
+            "table": table_name,
             "model": model,
             "table_creation_string": table_create_string,
             "references": references,
             "m2m_tables": m2m_tables_for_create,
         }
 
-    def _get_models_to_create(self, models_to_create: "list[type[Model]]") -> None:
+    def _get_models_to_create(self, models_to_create: list[type[Model]]) -> None:
```

Comment:
I think this function should just return the list of models instead of modifying the argument. I find it hard to understand why it should be modifying the argument instead...